csv/src/main/java/org/apache/metamodel/csv/SingleLineCsvDataSet.java

```diff
@@ -44,20 +44,56 @@
     private final ICSVParser _csvParser;
     private final int _columnsInTable;
     private final boolean _failOnInconsistentRowLength;
-
+    private final CsvConfiguration _csvConfiguration;
+    
     private volatile int _rowNumber;
     private volatile Integer _rowsRemaining;
     private volatile Row _row;
 
+    /**
+     * @param reader
+     * @param csvParser
+     * @param columns
+     * @param maxRows
+     * @param columnsInTable
+     * @param failOnInconsistentRowLength
+     * 
+     * @deprecated When instantiating a {@link SingleLineCsvDataSet} using this constructor the {@link ICSVParser} can
+     *             be reused in different threads in a parallel manner which is not promised to work by the
+     *             {@link ICSVParser}. Use
+     *             {@link #SingleLineCsvDataSet(BufferedReader, CsvConfiguration, List, Integer, int, boolean)} instead.
+     */
+    @Deprecated
```

Comment:
This class is not public however, so we don't even need to worry about deprecation etc. Just whack the old constructor.